app/menus/index.js

```diff
@@ -248,6 +249,95 @@ class Menus {
     const script = `document.querySelectorAll('video').forEach(v => {v.removeAttribute("disablepictureinpicture"); v.toggleAttribute("controls");})`;
     this.window.webContents.executeJavaScript(script, true);
   }
+
+  joinMeeting() {
+    try {
+      const clipboardText = clipboard.readText();
+      if (this.isValidTeamsMeetingUrl(clipboardText)) {
+        this.joinMeetingWithUrl(clipboardText);
+        return;
+      }
+    } catch (error) {
+      console.error('Error checking clipboard:', error);
+    }
+    
+    this.startClipboardMonitoring();
+  }
+
+  startClipboardMonitoring() {
+    if (this.clipboardMonitoring?.interval) {
+      return; // Already monitoring
+    }
+
+    this.clipboardMonitoring = {
+      interval: null,
+      lastUrl: null
+    };
+
+    this.clipboardMonitoring.interval = setInterval(() => {
+      this.checkClipboardForMeetingUrl();
+    }, 1000);
+
+    this.showWaitingForUrlDialog();
+  }
+
+  checkClipboardForMeetingUrl() {
+    try {
+      const clipboardText = clipboard.readText();
+      const isValidUrl = this.isValidTeamsMeetingUrl(clipboardText);
+      
+      if (isValidUrl && clipboardText !== this.clipboardMonitoring.lastUrl) {
+        this.clipboardMonitoring.lastUrl = clipboardText;
+        this.joinMeetingWithUrl(clipboardText);
+        this.stopClipboardMonitoring();
+      }
+    } catch (error) {
+      console.error('Error checking clipboard:', error);
+    }
+  }
+
+  isValidTeamsMeetingUrl(text) {
+    if (typeof text !== 'string') {
+      return false;
+    }
+    const teamsUrlPattern = /^https:\/\/teams\.microsoft\.com\/l\/meetup-join\//;
+    return teamsUrlPattern.test(text);
+  }
+
+  showWaitingForUrlDialog() {
+    dialog.showMessageBox(this.window, {
+      type: 'info',
+      title: 'Waiting for Teams Meeting URL',
+      message: 'Copy a Teams meeting URL to your clipboard',
+      detail: 'The app is monitoring your clipboard for a valid Teams meeting URL. Once you copy one, it will open automatically.',
+      buttons: ['Cancel'],
```

Comment:
if we provide a cancel button, it might be good to stop the monitoring on cancel, otherwise it will still be monitoring and you might end up opening a random teams url when you less expect it. Otherwise might be good to change the text to say "Close dialog" or similar